bindings/rust/s2n-tls/src/testing/s2n_tls.rs

```diff
@@ -757,4 +785,228 @@ mod tests {
+        let pair = Pair::new(server, client);
+        let pair = poll_tls_pair(pair);
+
+        let client = pair.client.0.connection;
+
+        // Check connection was full handshake and a session ticket was included
+        assert_eq!(
+            client.handshake_type().unwrap(),
+            "NEGOTIATED|FULL_HANDSHAKE|TLS12_PERFECT_FORWARD_SECRECY|WITH_SESSION_TICKET"
+        );
+
+        // create and configure a client/server connection again
+        let mut server = crate::connection::Connection::new_server();
+        server
+            .set_config(server_config)
+            .expect("Failed to bind config to server connection");
+
+        // create a client connection with a resumption ticket
+        let mut client = crate::connection::Connection::new_client();
+
+        client
+            .set_session(&*ticket_ptr.borrow())
+            .unwrap()
+            .set_config(client_config)
+            .expect("Unable to set client config");
+
+        let server = Harness::new(server);
+        let client = Harness::new(client);
+        let pair = Pair::new(server, client);
+        let pair = poll_tls_pair(pair);
+
+        let client = pair.client.0.connection;
+
+        // Check new connection was resumed
+        assert_eq!(client.handshake_type().unwrap(), "NEGOTIATED");
+    }
+
+    #[test]
+    fn resume_tls13_session() {
+        let keypair = CertKeyPair::default();
+
+        // Create test ticket key
+        let key: [u8; 3] = [1, 2, 3];
+        let keyname: [u8; 3] = [1, 3, 4];
+
+        // Initialize config for server with a ticket key
+        let mut server_config_builder = Builder::new();
+        server_config_builder
+            .enable_session_tickets()
+            .unwrap()
+            .add_session_ticket_key(&keyname, &key, 0)
+            .unwrap()
+            .load_pem(keypair.cert(), keypair.key())
+            .unwrap()
+            .set_security_policy(&security::DEFAULT_TLS13)
+            .unwrap();
+        let server_config = server_config_builder.build().unwrap();
+
+        // Creates session ticket callback handler
+        pub struct AcceptSessionTicketHandler {
+            stored_ticket: Rc<RefCell<Vec<u8>>>,
+        }
+
+        // Implement the session ticket callback
+        impl SessionTicketCallback for AcceptSessionTicketHandler {
+            fn on_session_ticket(
+                self: &AcceptSessionTicketHandler,
+                _connection: &mut crate::connection::Connection,
+                session_ticket: crate::session_ticket::SessionTicket,
+            ) {
+                if let Ok(data) = session_ticket.session_data() {
+                    (*self.stored_ticket).borrow_mut().extend(data);
+                }
+            }
+        }
+        let stored_ticket = RefCell::new(vec![]);
+        let ticket_ptr = Rc::new(stored_ticket);
+        let handler = AcceptSessionTicketHandler {
+            stored_ticket: ticket_ptr.clone(),
+        };
```

Comment:
I'd just derive `Default` on your `AcceptSessionTicketHandler`, then you don't have to manually create this.

Reply:
Does this actually work? I have to clone the ticket_ptr, and I don't think a derive default would do that?

Reply:
Yup, the default Rc<RefCell<Vec<u8>>> is an empty vec inside an Rc and RefCell.

Although while we're on the subject, I think I'd prefer this to be an option? `Option<Vec<u8>>` since the ticket is either there or not.